ft232/mpsse.py

```diff
@@ -0,0 +1,555 @@
+        """ Generic write command. 
+        Parameters
+        ----------
+        Data : list, bytes, bytearray
+            The data which should be send. Can be 1 to 65538 bytes long. Or 1 - 8 if data is read AS_BITS.
+        AsBitsBytes : MPSSE.AS_BITS / MPSSE.AS_BYTES
+            Weather a single byte of data is send or a list of bytes.
+        WriteClkEdge : MPSSE.WRITE_CLK_EDGE_POSITIVE / MPSSE.WRITE_CLK_EDGE_NEGATIVE
+            Weather the data should be send on positive or negative clock edge.
+        
+        BitOrder : MPSSE.BIT_ORDER_MSB / MPSSE.BIT_ORDER_LSB
+            Weather the data should be send MSB first or LSB first.
+        Tms : MPSSE.TMS_ON / MPSSE.TMS_OFF
+            Weather TMS is enabled/disabled.
+        """
+        Command = self.__dataCommand(MPSSE.WRITE_DATA, AsBitsBytes, WriteClkEdge, BitOrder, Tms)
+        if AsBitsBytes == MPSSE.AS_BITS:
+            Length = Data[0].bit_length() - 1
+            self.__Ft232.write(bytes([Command, Length]))
+        elif AsBitsBytes == MPSSE.AS_BYTES:
+            Length = len(Data) - 1
+            LengthLowByte = Length & 0xff
+            LengthHighByte = (Length >> 8) & 0xff
+            self.__Ft232.write(bytes([Command, LengthLowByte, LengthHighByte]))
+        self.__Ft232.write(bytes(Data))
+
+    def _read(self, ReadLength, AsBitsBytes, WriteClkEdge, BitOrder, Tms=TMS_OFF):
+        """ Generic read command. 
+        Parameters
+        ----------
+        ReadLength : int
+            The number of bytes or bits which should be read. Can be 1 to 65538 in bytes. Or 1 - 8 if data is read AS_BITS
+        AsBitsBytes : MPSSE.AS_BITS / MPSSE.AS_BYTES
+            Weather a number of bytes (AS_BYTES) is read or a number of bits (AS_BITS).
+        WriteClkEdge : MPSSE.WRITE_CLK_EDGE_POSITIVE / MPSSE.WRITE_CLK_EDGE_NEGATIVE
+            Weather the data should be read on positive (WRITE_CLK_EDGE_NEGATIVE) or negative (WRITE_CLK_EDGE_POSITIVE) clock edge. 
+        
+        BitOrder : MPSSE.BIT_ORDER_MSB / MPSSE.BIT_ORDER_LSB
+            Weather the data should be read MSB first or LSB first.
+        Tms : MPSSE.TMS_ON / MPSSE.TMS_OFF
+            Weather TMS is enabled/disabled.
+        Returns
+        -------
+        list(int)
+            A list with the read bytes.
+        """
+        Command = self.__dataCommand(MPSSE.READ_DATA, AsBitsBytes, WriteClkEdge, BitOrder, Tms)
+        if AsBitsBytes == MPSSE.AS_BITS:
+            ReadLength = ReadLength - 1
+            self.__Ft232.write(bytes([Command, ReadLength]))
+        elif AsBitsBytes == MPSSE.AS_BYTES:
+            ReadLength = ReadLength - 1
+            LenghtLowByte = ReadLength & 0xff
+            LenghtHighByte = (ReadLength >> 8) & 0xff
+            self.__Ft232.write(bytes([Command, LenghtLowByte, LenghtHighByte]))
+        return self.__Ft232.read(ReadLength)
+    
+    def _transfer(self, Data, ReadLength, AsBitsBytes, WriteClkEdge, BitOrder, Tms=TMS_OFF):
```

Comment:
It doesn't make much sense to me to have the MPSE class to exist with only a mostly private interface only for it to be wrapped in higher layers. I think the use of classes in the patchset is overly complex...